src/commands/claim-nii.js

```diff
@@ -0,0 +1,109 @@
+
+        let spinner = ora();
+        try {
+            const RevenueTokenManagerContract = require('../contracts/revenue-token-manager-contract');
+            const revenueTokenManager = new RevenueTokenManagerContract(wallet);
+
+            let niiBalance = ethers.utils.formatUnits(await niiContract.balanceOf(config.wallet.address), 15);
+            dbg(`Opening on-chain balance: ${niiBalance} NII`);
+
+            spinner.start(`1/6 - Registering claim for period ${period}`);
+            const releaseTx = await revenueTokenManager.release(period - 1, options);
+            spinner.succeed(`1/6 - Claim registered for period ${period}`);
+
+            spinner.start('2/6 - Confirming claim');
+            const releaseReceipt = await provider.getTransactionConfirmation(releaseTx.hash);
+            spinner.succeed('2/6 - Claim confirmed');
+
+            niiBalance = ethers.utils.formatUnits(await niiContract.balanceOf(config.wallet.address), 15);
+            dbg(`Depositing: ${niiBalance} NII`);
+
+            spinner.start(`3/6 - Approving transfer of ${niiBalance} NII`);
+            const pendingApprovalTx = await wallet.approveTokenDeposit(niiBalance, 'NII', options);
+            spinner.succeed('3/6 - Transfer approval registered');
+
+            spinner.start('4/6 - Confirming transfer approval');
+            const approveReceipt = await provider.getTransactionConfirmation(pendingApprovalTx.hash);
+            spinner.succeed('4/6 - Transfer approval confirmed');
+
+            spinner.start('5/6 - Registering nahmii deposit');
+            const pendingCompleteTx = await wallet.completeTokenDeposit(niiBalance, 'NII', options);
+            spinner.succeed('5/6 - nahmii deposit registered');
+
+            spinner.start('6/6 - Confirming nahmii deposit');
+            const completeReceipt = await provider.getTransactionConfirmation(pendingCompleteTx.hash);
+            spinner.succeed(`6/6 - nahmii deposit of ${niiBalance} NII confirmed`);
+
+            console.error('Please allow a few minutes for the nahmii balance to be updated!');
+
+            const output = [releaseReceipt, approveReceipt, completeReceipt].map(reduceReceipt);
+            console.log(JSON.stringify(output));
+        }
+        catch (err) {
+            dbg(err);
+            spinner.fail();
+            throw new Error(`Claiming NII failed: ${err.message}`);
+        }
+        finally {
+            let niiBalance = ethers.utils.formatUnits(await niiContract.balanceOf(config.wallet.address), 15);
+            dbg(`Closing on-chain balance: ${niiBalance} NII`);
+
+            provider.stopUpdate();
+        }
+    }
+};
+
+let validateGasLimitIsPositiveInteger = function(gas) {
+    const gasLimit = parseInt(gas);
+    if (gasLimit <= 0)
+        throw new Error('Gas limit must be a number higher than 0');
+    return gasLimit;
+};
+
+let validatePeriodIsPositiveInteger = function(period) {
```

Comment:
I think `validatePeriod` would be a more accurate name for this function. 

Also, any particular reason why you're not using `const`?

Reply:
Copy paste syndrome.. changed to regular named functions.